connect_ext_test.go

```diff
@@ -575,6 +575,40 @@ func Test_CustomCompression(t *testing.T) {
 	assert.Equal(t, response.Msg, &pingv1.PingResponse{Text: request.Text})
 }
 
+func TestConnectTimeout(t *testing.T) {
```

Comment:
Should this be called `TestInvalidHeaderTimeout` since we are not testing the timeout itself, but more malformed headers?